scvi/model/scanvi.py

```diff
@@ -259,6 +259,10 @@ def train(
         self.trainer.labelled_set = self.trainer.create_scvi_dl(
             indices=self._labeled_indices
         )
+
+        for scdl in [self.trainer.labelled_set, self.trainer.unlabelled_set]:
+            scdl.to_monitor = ["reconstruction_error", "accuracy"]
+
```

Comment:
isn't this already done?

https://github.com/YosefLab/scvi-tools/blob/b519575a5469f1ec71250e06a8856e155bbf5c91/scvi/core/trainers/annotation.py#L187-L188

Reply:
The problem is that it is done in trainer and then scanvi assigns datasets after trainer init without `to_monitor`.

Reply:
lol, we really need to refactor all the semisupervised trainers